tests/system/python/test_end_to_end_csv.py

```diff
@@ -0,0 +1,243 @@
+
+    conn = http.client.HTTPConnection(foglamp_url)
+    subprocess.run(["rm -rf /tmp/foglamp-south-{}".format(south_plugin)], shell=True, check=True)
+    subprocess.run(["rm -rf -d ${FOGLAMP_ROOT}/python/foglamp/plugins/south/*/"], shell=True, check=True)
+    subprocess.run(["git clone https://github.com/foglamp/foglamp-south-{}.git /tmp/foglamp-south-{}".
+                   format(south_plugin, south_plugin)], shell=True, check=True)
+    subprocess.run(["cp -r /tmp/foglamp-south-{}/python/foglamp/plugins/south/* "
+                    "$FOGLAMP_ROOT/python/foglamp/plugins/south/".format(south_plugin)], shell=True, check=True)
+    subprocess.run(["rm -rf $FOGLAMP_ROOT/data/{}".format(CSV_NAME)], shell=True, check=True)
+    csv_file_path = os.path.join(os.path.expandvars('${FOGLAMP_ROOT}'), 'data/{}'.format(CSV_NAME))
+    f = open(csv_file_path, "w")
+    f.write(CSV_HEADER)
+    f.write("\n{}".format(CSV_DATA))
+    f.close()
+
+    conn.request("POST", '/foglamp/service', json.dumps(data))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    retval = json.loads(r)
+    assert retval["name"] == "play"
+    return csv_file_path
+
+
+def _start_folamp_north(foglamp_url, pi_host, pi_port, north_plugin, pi_token):
+    """Start north task"""
+
+    conn = http.client.HTTPConnection(foglamp_url)
+    data = {"name": "North_Readings_to_PI",
+            "plugin": "{}".format(north_plugin),
+            "type": "north",
+            "schedule_type": 3,
+            "schedule_day": 0,
+            "schedule_time": 0,
+            "schedule_repeat": 30,
+            "schedule_enabled": "false"}
+    conn.request("POST", '/foglamp/scheduled/task', json.dumps(data))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r = r.read().decode()
+    retval = json.loads(r)
+    assert retval["name"] == "North_Readings_to_PI"
+
+    # Test the north is setup and configuration is available
+    retry_count = 0
+    config_status = None
+    while config_status != 200 and retry_count < NUM_RETRIES:
+        conn.request("GET", '/foglamp/category/North_Readings_to_PI/producerToken')
+        r = conn.getresponse()
+        config_status = r.status
+        r.read().decode()
+        retry_count += 1
+        time.sleep(SLEEP_INTERVAL_LOW)
+
+    if config_status != 200 or retry_count == NUM_RETRIES:
+        assert False, "North plugin config load failed in {} tries".format(NUM_RETRIES)
+
+    data = {"URL": "https://{}:{}/ingress/messages".format(pi_host, pi_port), "producerToken": "{}".format(pi_token)}
+    conn.request("PUT", '/foglamp/category/North_Readings_to_PI', json.dumps(data))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r.read().decode()
+
+    data = {"schedule_name": "{}".format("North_Readings_to_PI")}
+    conn.request("PUT", '/foglamp/schedule/enable', json.dumps(data))
+    r = conn.getresponse()
+    assert 200 == r.status
+    r.read().decode()
```

Comment:
We can do enable task with good config in **2 steps**:

1) Fetch north plugins with their default config and get north_plugin key from the JSON => 

> http://localhost:8081/foglamp/plugins/installed?type=north&config=true

2) Update the config from point1 and  then add north task with updated config via below
> curl -sX POST http://localhost:8081/foglamp/scheduled/task -d
     '{"name": "PI-2",
     "plugin": "pi_server",
     "type": "north",
     "schedule_type": 3,
     "schedule_day": 0,
     "schedule_time": 0,
     "schedule_repeat": 30,
     "schedule_enabled": true,
     "config": {
        "producerToken": {"value": "uid=180905062754237&sig=kx5l+"},
        "URL": {"value": "https://10.2.5.22:5460/ingress/messages"}}}'
    """